cmd/spx/template/game.js

```diff
@@ -261,27 +257,25 @@ class GameApp {
         this.hasMergedProject = true
     }
 
-    async installProject() {
+    async startProject() {
         try {
+            this.onProgress(0.1);
+            this.editor = new Engine(this.editorConfig);
+            this.clearPersistence(this.tempZipPath);
             let dbExists = await this.checkDBExist(this.persistentPath, this.getInstallPath());
             console.log(this.getInstallPath(), " DBExist result= ", dbExists, "this.isEditor", this.isEditor);
-            if (dbExists && !this.isEditor) {
-                this.RunGame();
+            if (!dbExists) {
+                // install project
+                // TODO store project zip and hash to indexDB 
+                this.exitFunc = this.runEditor.bind(this);
+                this.editor.init(this.basePath).then(async () => {
+                    await this.mergeProjectWithEngineRes()
+                    this.editor.copyToFS(this.tempZipPath, this.projectData);
+                    const args = ['--project-manager', '--single-window', "--install_project_name", this.projectInstallName];
+                    this.editor.start({ 'args': args, 'persistentDrops': true });
```

Comment:
Should we call `startProjectResolve` when `this.editor.start()` here finishes?

Reply:
```js
  await this.mergeProjectWithEngineRes()
  this.editor.copyToFS(this.tempZipPath, this.projectData);
  const args = ['--project-manager', '--single-window', "--install_project_name", this.projectInstallName];
  this.editor.start({ 'args': args, 'persistentDrops': true });
```
They are just installing the project to web IndexDB. We need to call `runEditor` to verify and import assets, so we should call `startProjectResolve` after `runEditor` finished.

```js
this.exitFunc = this.runEditor.bind(this);
```